Add OfflineTools and unit tests (#27)
kuzzleio/sdk-csharp: doc/1/offline/offline-manager/setters/index.md

---
code: true
type: page
title: Setters
order: 100
desription: Setters for Offline
---

> # OfflineManager class setters

Reviewer: ```suggestion
# OfflineManager class properties
```

Reviewer: Also, you should add the default values for all these properties.

Reviewer: One last thing: properties should be documented in alphabetical order.

## MaxQueueSize

> Return the maximum amount of elements that the offline queue can contains.

Reviewer: Since this is not only a getter, you cannot describe this with `return`. Instead, I think the following form is more adapted:

```suggestion
Gets or sets the maximum amount of elements that the offline queue can contain.
```

Reviewer: I just saw that you separated getters and setters, and I don't see why: getters and setters are only properties of a property, and so they shouldn't be documented themselves. 

What's important for SDK users is the underlying property, what it does, what's its default value, and what is the impact of changing it.

### Signature

```csharp
public int MaxQueueSize {
    get { return maxQueueSize; }

>     set { maxQueueSize = value < 0 ? -1 : value; }

Reviewer: I'm not fond of putting implementation details in the documentation, as it prevents us from changing it if needed, because this causes a breaking change.

I think that something like that is preferable:

    ### Definition

    ```csharp
    public int MaxQueueSize { get; set; }
    ```

(to be applied to all similarly documented properties)

}
```

## MinTokenDuration

> Return the minimum duration of a Token after refresh.

Reviewer: This is a bit weird, because if there is a minimum, then what is the maximum?

This property might be better named `RefreshedTokenDuration`?

Also, typo:

```suggestion
Gets or sets the minimum duration of a token after being refreshed.
```

### Signature

```csharp
public int MinTokenDuration {
    get { return minTokenDuration; }
    set { minTokenDuration = value < 0 ? -1 : value; }
}
```

## MaxRequestDelay

> Return the maximum delay between two requests to be replayed

Reviewer: ```suggestion
Gets or sets the maximum delay between two replayed requests.
```

### Signature

```csharp
public int MaxRequestDelay {
    get { return maxRequestDelay; }
    set { maxRequestDelay = value < 0 ? 0 : value; }
}
```

## QueueFilter

> Return the queue filter applied when a queue is being replayed.

Reviewer: ```suggestion
Gets or sets the custom queue filter applied when a queue is being replayed.
```

### Signature

```csharp
public Func<JObject, bool> QueueFilter { get; set; }
```

## Usage

> <<< ./snippets/setters.cs

Reviewer: I don't think that a code example is relevant for class properties.
